app/gui/src/presenter/searcher.rs

```diff
@@ -346,6 +346,18 @@ impl Searcher {
 
             action_list_changed <- source::<()>();
             select_entry <- action_list_changed.filter(f_!(model.should_auto_select_first_action()));
+
+            eval_ model.view.toggle_private_cb_entries_visibility (model.controller.reload_list());
+            action_list_loaded <-
+                action_list_changed.map(f_!([model] !model.controller.actions().is_loading()));
+            action_list_loaded <- action_list_loaded.on_true();
+            reloading_action_list <-
+                bool(&action_list_loaded, &model.view.toggle_private_cb_entries_visibility);
+            reloading_action_list <- reloading_action_list.on_change();
+            reloading_action_list_done <- reloading_action_list.on_false();
+            searcher_input_on_visibility_toggle <-
+                model.view.searcher_input_changed.sample(&reloading_action_list_done);
+            eval searcher_input_on_visibility_toggle ((expr) model.input_changed(expr));
```

Comment:
Why do we need to reset searcher input in controllers after a reload?

Reply:
Yup, it looks strange and performance-hurting. Good catch @farmaazon!

Reply:
After reloading the component list we need to update the filtering. Otherwise, all components will be shown after a toggle, irrespective of them matching the searcher input pattern.

I've refactored this to only call the `update_filtering` methods and not set a new searcher input, as indeed the searcher input does not change.